src/model/index.py

```diff
@@ -0,0 +1,74 @@
+import os
+import json
+import pathlib
+from output.exceptions import *
+from commands.schema_keys import *
+
+
+class Index:
+    def __init__(self, index_name, table_metadata):
+        Index.__validate_index__(index_name, table_metadata.columns)
+        self.name = index_name
+        self.__path__ = os.path.join(table_metadata.get_path(), self.name)
+
+    def get_path(self):
+        return self.__path__
+
+    @staticmethod
+    def __validate_index__(index_name, table_columns):
+        if index_name not in table_columns:
+            raise WrongParameterError("Index {} not found".format(index_name))
+
+    @staticmethod
+    def __validate_value_name__(value_name, path):
+        if len(value_name) == 0 or value_name.isspace():
+            raise NoParameterError("value_name parameter not entered")
+        if not os.path.exists(os.path.join(path, "{}.json".format(value_name))):
+            raise WrongParameterError("{} doesn't exist".format(value_name))
+
+    @staticmethod
+    def __validate_primary_keys__(primary_keys):
+        if primary_keys is None:
+            raise NoParameterError("primary_keys parameter not entered")
+
+    @staticmethod
+    def __validate_primary_key__(primary_key):
+        if len(primary_key) == 0 or primary_key.isspace():
+            raise NoParameterError("primary_key parameter not entered")
+
+    def serialize(self):
+        os.makedirs(self.__path__, exist_ok=True)
+
+    def get_primary_keys(self, value_name):
+        Index.__validate_value_name__(value_name, self.path)
+        with open(os.path.join(self.path, "{}.json".format(value_name)), 'r') as file:
+            return json.load(file)
+
+    @staticmethod
+    def __update_value__(path, value_name, primary_keys):
+        Index.__validate_value_name__(value_name, path)
+        Index.__validate_primary_keys__(primary_keys)
+        with open(os.path.join(path, "{}.json".format(value_name)), 'w') as file:
+            json.dump(primary_keys, file)
+
+    def add_value(self, value_name, primary_key):
+        Index.__validate_value_name__(value_name, self.path)
+        Index.__validate_primary_key__(primary_key)
+        path = os.path.join(self.path, "{}.json".format(value_name))
```

Comment:
if we create a first entry in the table student 
`{"id" : "1",  "first-name", "yahya"}`. forsure, file `yehya,json` won't exist. so we shouldn't throw an exception and create it. 

we don't need a `__validate_value_name` to check if the file exists or not